ibc/ibc_handle_event_test.go

```diff
@@ -0,0 +1,242 @@
+package ibc
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/pokt-network/pocket/persistence/kvstore"
+	"github.com/pokt-network/pocket/shared/messaging"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHandleEvent_FlushCaches(t *testing.T) {
+	mgr, _, _, _, ibcMod := prepareEnvironment(t, 1, 0, 0, 0)
+	tmpDir := mgr.GetConfig().IBC.StoresDir + "/caches"
+	ibcHost := ibcMod.GetBus().GetIBCHost()
+	store, err := ibcHost.GetProvableStore("test")
+	require.NoError(t, err)
+
+	// set height
+	publishNewHeightEvent(t, ibcMod.GetBus(), 1)
+
+	kvs := []struct {
+		key   []byte
+		value []byte
+	}{
+		{
+			key:   []byte("key1"),
+			value: []byte("value1"),
+		},
+		{
+			key:   []byte("key2"),
+			value: []byte("value2"),
+		},
+		{
+			key:   []byte("key3"),
+			value: []byte("value3"),
+		},
+		{
+			key:   []byte("key4"),
+			value: nil,
+		},
+	}
+
+	for _, kv := range kvs {
+		if kv.value != nil {
+			require.NoError(t, store.Set(kv.key, kv.value))
+		} else {
+			require.NoError(t, store.Delete(kv.key))
+		}
+	}
+
+	// increment the height
+	publishNewHeightEvent(t, ibcMod.GetBus(), 2)
+
+	cache, err := kvstore.NewKVStore(tmpDir)
+	require.NoError(t, err)
+	keys, values, err := cache.GetAll([]byte{}, false)
+	require.NoError(t, err)
+	require.Len(t, keys, 4)
+	require.Len(t, values, 4)
+	for i, key := range keys {
+		require.Equal(t, string(key), prepareCacheKey(1, kvs[i].key))
+		if kvs[i].value == nil {
+			require.Equal(t, values[i], []byte{})
+			continue
+		}
+		require.Equal(t, values[i], kvs[i].value)
+	}
+
+	err = cache.ClearAll()
+	require.NoError(t, err)
+
+	newKeys, newValues, err := cache.GetAll([]byte{}, false)
+	require.NoError(t, err)
+	require.Len(t, newKeys, 0)
+	require.Len(t, newValues, 0)
+
+	require.NoError(t, cache.Stop())
+
+	// flush the cache
```

Comment:
You only flush AFTER you've cleared everything.

Do you not want to:

1. Store
2. Flush
3. Restore
4. Validate (non empty)

5. Clear

6. Flush
7. Restore
8. Validate (empty)

Reply:
Yeah I am flushing to confirm the original flush got everything. So I take the first flush, confirm it flushed the correct amount, clear the disk cache and then flush again without any updates to ensure nothing was left. Your method would require I restore and flush again as you cannot access the cache in memory

Reply:
> clear the disk cache and then flush again without any updates to ensure nothing was left

So is this part of the test or is it a "cleanup" task. If it's a cleanup, use `t.cleanup` (and update other parts of the code as well). This way it's clear that we're not testing flushing, but simply have tests that are dependant on each other.

Reply:
Its not cleanup, its ensuring everything was cached the first time round. cleanup is just stopping the store and main_test.go deletes the directory when done

Reply:
If the second flush put anything in the cache we would know the first cache job didnt work correctly its a double verification